File: BMO2/BMO2/frmPrincipal.vb
Imports System.Threading
Public Class frmPrincipal
    Dim _loop As Integer = 0
    Private Sub BtnComenzar_Click(sender As Object, e As EventArgs) Handles btnComenzar.Click
        _loop = 10
        bgwBarra.RunWorkerAsync()
    End Sub

    Private Sub BgwBarra_DoWork(sender As Object, e As System.ComponentModel.DoWorkEventArgs) Handles bgwBarra.DoWork
        For x As Integer = 0 To _loop
            Thread.Sleep(1000)
            bgwBarra.ReportProgress(CInt((x / _loop) * 100))

        Next
    End Sub

    Private Sub BgwBarra_ProgressChanged(sender As Object, e As System.ComponentModel.ProgressChangedEventArgs) Handles bgwBarra.ProgressChanged
        Me.prgBarra.Value = e.ProgressPercentage
    End Sub

    Private Sub BgwBarra_RunWorkerCompleted(sender As Object, e As System.ComponentModel.RunWorkerCompletedEventArgs) Handles bgwBarra.RunWorkerCompleted
        frmMenu.Show()
        Me.Close()
    End Sub
End Class

File: BMO/BMO/frmPrincipal.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Threading;

namespace BMO
{
    public partial class frmPrincipal : Form
    {
        Thread Me;
        public frmPrincipal()
        {
            InitializeComponent();
        }

        private void BtnComenzar_Click(object sender, EventArgs e)
        {
            if (bgwBarra.IsBusy != true)
            {
                bgwBarra.RunWorkerAsync();
            }
        }

        private void BgwBarra_DoWork(object sender, DoWorkEventArgs e)
        {
            for (int i = 1; i <= 10; i++)
            {
                Thread.Sleep(1000);
                bgwBarra.ReportProgress(i * 10);
            }
        }

        private void BgwBarra_ProgressChanged(object sender, ProgressChangedEventArgs e)
        {
            prgBarra.Value = e.ProgressPercentage;
        }

        private void BgwBarra_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            this.Close();
            Me = new Thread(Otro);
            Me.SetApartmentState(ApartmentState.STA);
            Me.Start();
        }

        private void Otro(object obj)
        {
            Application.Run(new frmMenu());
        }
    }
}
